doc/migrating_to_spire_agent_id_enforced_format.md

```diff
@@ -0,0 +1,101 @@
+# SPIRE Agent ID Migration Guide
+
+This guide outlines the steps required to update your SPIRE agent IDs to conform to the enforced ID shape.
```

Comment:
We should use "SPIRE Agent" instead of "SPIRE agent".
```suggestion
This guide outlines the steps required to update your SPIRE Agent IDs to conform to the enforced ID shape.
```